### PR TITLE
Installing the ssh2 Package for Zowe CLI

### PR DESCRIPTION
**Pull request by aimmta**
Signed-off-by: Michael Aimino <michael.aimino@broadcom.com>

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../docs/contribute/contributing.md) to this repository.

- [ ] If the changes in this PR is part of the next future release, make this pull request against the **docs-staging** branch which will be published at the next release boundary. If the changes in this PR are part of the current release, use the default base branch, **master**. For more information about branches, see https://github.com/zowe/docs-site/tree/master#understanding-the-doc-branches. 
      
- [ ] If this PR relates to GitHub issues in `docs-site` or other repositories, please list in Description, prefixed with **close**, **fix** or **resolve** keywords.

### Description (including links to related git issues)
Updates to Zowe CLI installation docs, and new topic, to support ssh2 optional pre-req.

:heart:Thank you!

### CODE REVIEW

File: docs/user-guide/cli-installcli.md [hunk 1]
```diff
@@ -8,8 +8,10 @@ Install Zowe&trade; CLI on your computer.
 
 Use one of the following methods to install Zowe CLI.
 
-- [Installing Zowe CLI from a local package](#installing-zowe-cli-from-a-local-package)
-- [Installing Zowe CLI from an online registry](#installing-zowe-cli-from-an-online-registry)
+- [Installing Zowe CLI](#installing-zowe-cli)
+  - [Methods to install Zowe CLI](#methods-to-install-zowe-cli)
```
Comment (awharn) [inline docs/user-guide/cli-installcli.md:12]: Should this list include the two headings from above?

File: docs/user-guide/cli-installcli.md [hunk 2]
```diff
@@ -35,6 +37,8 @@ If you do not have internet access at your site, use the following method to ins
   Npm is included with most Node.js installations. Issue the command `npm --version` to check your current version. You can reference the [Node.js release matrix](https://nodejs.org/en/download/releases/) to verify that the versions are compatible.
   :::
 
+- **(Optional) ssh2 Package:** Zowe CLI has a dependency on the ssh2 package. This package allows for more secure cryptographic ciphers to be used first on supporting hardware. See [Installing the ssh2 Package for Zowe CLI](cli-installing-ssh2-package.md).
```
Comment (awharn) [inline docs/user-guide/cli-installcli.md:40]: This optional step should probably also be present in the other method to install Zowe CLI (around line 104)

File: docs/user-guide/cli-installing-ssh2-package.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
+# Installing the ssh2 Package for Zowe CLI
+​
+This topic describes how to install the ssh2 package to support Zowe CLI.
```
Comment (awharn) [inline docs/user-guide/cli-installing-ssh2-package.md:3]: This describes how to install the pre-requisites to enable optional features in the ssh2 package.

File: docs/user-guide/cli-installing-ssh2-package.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+- `make`
+- A C/C++ Compiler (i.e. `gcc-c++`, `g++`)
+
+## MacOS
+- Python 3.7 or greater
+- [CMake](https://cmake.org/download/)
+- XCode
+  - Including `XCode Command Line Tools`
+
+## API Examples
```
Comment (awharn) [inline docs/user-guide/cli-installing-ssh2-package.md:36]: The API Examples and below are not required for this documentation. These are for using the z/OS USS SDK.

**Review by awharn — CHANGES_REQUESTED** [review]
Leaving a couple recommended changes and observations.

File: docs/user-guide/cli-installcli.md (hunk 1 above)
Comment (aimmta) [reply docs/user-guide/cli-installcli.md:12]: No, some toc-related code got deleted in the md file. Fixed.

**Pull request merged**